sdk/src/wallet/operations/transaction/high_level/delegation/delay.rs

```diff
@@ -0,0 +1,110 @@
+
+use crate::{
+    client::{api::PreparedTransactionData, secret::SecretManage},
+    types::block::{
+        context_input::{CommitmentContextInput, ContextInput},
+        output::{AddressUnlockCondition, BasicOutput, DelegationId, DelegationOutputBuilder, MinimumOutputAmount},
+    },
+    wallet::{operations::transaction::TransactionOptions, types::TransactionWithMetadata, Wallet},
+};
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    crate::wallet::Error: From<S::Error>,
+    crate::client::Error: From<S::Error>,
+{
+    /// Delay a delegation's claiming. The `reclaim_excess` flag indicates whether excess value over the minimum storage
+    /// requirements will be moved to a basic output that is unlockable by the same address which controls the
+    /// delegation. Otherwise it will be added to a new delegation. In this manner, one can delegate for one epoch
+    /// at a time and never lose out on any rewards.
+    pub async fn delay_delegation_claiming(
+        &self,
+        delegation_id: DelegationId,
+        reclaim_excess: bool,
+    ) -> crate::wallet::Result<TransactionWithMetadata> {
+        let prepared_transaction = self
+            .prepare_delay_delegation_claiming(delegation_id, reclaim_excess)
+            .await?;
+
+        self.sign_and_submit_transaction(prepared_transaction, None, None).await
+    }
+
+    /// Prepare to delay a delegation's claiming. The `reclaim_excess` flag indicates whether excess value
+    /// over the minimum storage requirements will be moved to a basic output that is unlockable by the same address
+    /// which controls the delegation.
+    /// Otherwise it will be added to a new delegation. In this manner, one can delegate for one epoch at a time and
+    /// never lose out on any rewards.
+    pub async fn prepare_delay_delegation_claiming(
+        &self,
+        delegation_id: DelegationId,
+        reclaim_excess: bool,
+    ) -> crate::wallet::Result<PreparedTransactionData> {
+        let delegation_output = self.get_delegation_output(delegation_id).await.ok_or_else(|| {
+            crate::wallet::Error::DelegationTransitionFailed(format!(
+                "no delegation output found with id {delegation_id}"
+            ))
+        })?;
+        let protocol_parameters = self.client().get_protocol_parameters().await?;
+        // Add a commitment context input with the latest slot commitment
+        let slot_commitment_id = self.client().get_info().await?.node_info.status.latest_commitment_id;
+        let context_input = ContextInput::from(CommitmentContextInput::new(slot_commitment_id));
+
+        let min_delegation_amount = delegation_output
+            .output
+            .minimum_amount(protocol_parameters.storage_score_parameters());
+
+        // In order to split the output, the amount must be at least twice the minimum for a delegation output
+        let can_split = delegation_output.output.amount() >= 2 * min_delegation_amount;
```

Comment:
Should be `min_delegation_amount + min_basic_amount` instead?
So maybe just put the `min_basic_amount` definition from L63 above and use this value?

Reply:
Ah true, was confused by the `BasicOutput::minimum_amount()`, how about moving this line to L75?

Reply:
moved

Reply:
Thanks